File: src/main/java/org/mitallast/queue/rest/transport/HttpServer.java
package org.mitallast.queue.rest.transport;

import com.google.inject.Inject;
import com.typesafe.config.Config;
import io.netty.channel.ChannelInitializer;
import io.netty.channel.socket.SocketChannel;
import org.mitallast.queue.common.netty.NettyServer;
import org.mitallast.queue.rest.RestController;

public class HttpServer extends NettyServer {
    private RestController restController;

    @Inject
    public HttpServer(Config config, RestController restController) {
        super(config.getConfig("rest"), HttpServer.class);
        this.restController = restController;
    }

    @Override
    protected ChannelInitializer<SocketChannel> channelInitializer() {
        return new HttpServerInitializer(new HttpServerHandler(restController));
    }
}

File: src/main/java/org/mitallast/queue/rest/netty/codec/server/HttpServer.kt
package org.mitallast.queue.rest.netty.codec.server

import com.google.inject.Inject
import com.typesafe.config.Config
import io.netty.channel.ChannelInitializer
import io.netty.channel.socket.SocketChannel
import org.mitallast.queue.common.netty.NettyProvider
import org.mitallast.queue.common.netty.NettyServer

class HttpServer @Inject constructor(
    config: Config,
    provider: NettyProvider,
    private val serverHandler: HttpServerHandler
) :
    NettyServer(config, provider, config.getString("rest.custom.host"), config.getInt("rest.custom.port")) {

    override fun channelInitializer(): ChannelInitializer<SocketChannel> {
        return HttpServerInitializer(serverHandler)
    }
}
